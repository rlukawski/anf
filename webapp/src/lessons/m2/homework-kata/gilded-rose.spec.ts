import { Item, GildedRose } from "./gilded-rose";

describe("GIVEN Gilded Rose is initialized", function () {
  let glideRose: GildedRose;
  beforeAll(() => {
    glideRose = new GildedRose([
      new Item("Magic", 10, 10),
      new Item("Aged Brie", 10, 10),
      new Item("Backstage passes to a TAFKAL80ETC concert", 10, 10),
      new Item("Sulfuras, Hand of Ragnaros", 10, 10),
    ]);
  });
  describe("WHEN items are updated", function () {
    let items: Array<Item>;
    beforeAll(() => {
      items = glideRose.updateQuality();
    });

    test('to match first snapshot', () => {
      expect(items).toMatchSnapshot();
    })   
  });

  describe("WHEN items are updated second time ", function () {
    let items: Array<Item>;
    beforeAll(() => {
      items = glideRose.updateQuality();
    });

    test('to match second snapshot', () => {
      expect(items).toMatchSnapshot();
    })   
  });  
});
