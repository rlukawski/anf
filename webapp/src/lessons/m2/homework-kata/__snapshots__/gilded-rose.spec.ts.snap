// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN Gilded Rose is initialized WHEN items are updated second time  to match second snapshot 1`] = `
Array [
  Item {
    "name": "Magic",
    "quality": 8,
    "sellIn": 8,
  },
  Item {
    "name": "Aged Brie",
    "quality": 12,
    "sellIn": 8,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 14,
    "sellIn": 8,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 10,
    "sellIn": 10,
  },
]
`;

exports[`GIVEN Gilded Rose is initialized WHEN items are updated to match first snapshot 1`] = `
Array [
  Item {
    "name": "Magic",
    "quality": 9,
    "sellIn": 9,
  },
  Item {
    "name": "Aged Brie",
    "quality": 11,
    "sellIn": 9,
  },
  Item {
    "name": "Backstage passes to a TAFKAL80ETC concert",
    "quality": 12,
    "sellIn": 9,
  },
  Item {
    "name": "Sulfuras, Hand of Ragnaros",
    "quality": 10,
    "sellIn": 10,
  },
]
`;
